components/econet/econet.h

```diff
@@ -74,7 +77,23 @@ class Econet : public Component, public uart::UARTDevice {
 
   void set_src_address(uint32_t address) { src_adr_ = address; }
   void set_dst_address(uint32_t address) { dst_adr_ = address; }
-  void set_update_interval(uint32_t interval_millis) { update_interval_millis_ = interval_millis; }
+  void set_mod_req_updates_(std::vector<uint8_t> keys, std::vector<uint32_t> values) {
+    for (auto i = 0; i < keys.size(); i++) {
+      request_mod_interval_millis_[keys[i]] = values[i];
+      request_mod_last_requested_[i] = i * values[i] / MAX_REQUEST_MODS;
+      min_update_interval_millis_ = std::min(min_update_interval_millis_, values[i]);
+    }
+    min_delay_between_read_requests_ = update_interval_millis_ / MAX_REQUEST_MODS;
```

Comment:
What's the benefit pre-computing this here? The way you have it you have to compute it twice in two different setters. It's a simple division. I'd just compute this every time you need it. The advantage is there you can divide by the actual number of request mods used and being able to spread them out even more.
The only thing that is slightly worth precomputing is the minimum value across all request_mod_update_intervals. Then in a private function compute the min delay as min(min across request_mod_update_intervals, update_interval) / request_mods_

Reply:
In looking at this I think it makes sense to move this to the same place request_mods_ gets set insider register_listener:

```
    request_mods_ = std::max(request_mods_, (uint8_t) (request_mod + 1));
    min_delay_between_read_requests_ = std::max(min_update_interval_millis_ / request_mods_,200);
```